File: Day6/C/Ques22.c
#include <stdio.h>
#include <math.h>
int main()
{
    int bin, dec = 0, i = 0;
    printf("Enter a binary number:");
    scanf("%d", &bin);
    int b = bin;
    while (bin > 0)
    {
        int a = bin % 10;
        int t = a * pow(2, i);
        dec = dec + t;
        i++;
        bin = bin / 10;
    }
    printf("%d into Decimal: %d", b, dec);
    return 0;
}

File: Day6/Python/Ques22.py
import math
n=int(input("Enter a binary number:"))
dec=0
i=t=0
while n>0:
    t=n%10 * math.pow(2,i)
    dec += t
    i +=1
    n=n//10
print("The binary to decimal conversion of the given number is:",dec)
